scope_test.go

```diff
@@ -435,3 +436,181 @@ func TestDecorateRequest(t *testing.T) {
 		}
 	}
 }
+
+func TestGetHostSticky(t *testing.T) {
+	c := &cluster{
+		name:     "default",
+		replicas: []*replica{{}, {}, {}},
+	}
+	r1 := c.replicas[0]
+	r1.cluster = c
+	r1.hosts = []*host{
+		{
+
+			addr:    &url.URL{Host: "127.0.0.11"},
+			active:  1,
+			replica: r1,
+		},
+		{
+			addr:    &url.URL{Host: "127.0.0.22"},
+			active:  1,
+			replica: r1,
+		},
+	}
+	r1.name = "replica1"
+	r2 := c.replicas[1]
+	r2.cluster = c
+	r2.hosts = []*host{
+		{
+			addr:    &url.URL{Host: "127.0.0.33"},
+			active:  1,
+			replica: r2,
+		},
+		{
+			addr:    &url.URL{Host: "127.0.0.44"},
+			active:  1,
+			replica: r2,
+		},
+	}
+	r2.name = "replica2"
+	r3 := c.replicas[2]
+	r3.cluster = c
+	r3.hosts = []*host{
+		{
+			addr:    &url.URL{Host: "127.0.0.55"},
+			active:  1,
+			replica: r3,
+		},
+		{
+			addr:    &url.URL{Host: "127.0.0.66"},
+			active:  1,
+			replica: r3,
+		},
+	}
+	r3.name = "replica3"
+	exceptedSessionHostMap := map[string]string{
+		"0": c.getHostSticky("0").addr.Host,
+		"1": c.getHostSticky("1").addr.Host,
+		"2": c.getHostSticky("2").addr.Host,
+		"3": c.getHostSticky("3").addr.Host,
```

Comment:
since you rely in expectations on the same function that computes results, it will not be resilient to mistakes in `getHostSticky`. Could you rather hardcode expectations (expected sticky hosts)? 
Please do the same for other tests